readme.md

```diff
@@ -39,6 +39,14 @@ details.branchName // is null if the repository in detached HEAD mode
 details.isCleanTag
 ```
 
+You can optionally search a subset of tags with `prefix`. Inclusion of the `prefix` in the returned version is toggled
+with `includePrefix` (default: `false`). Example when the tag is my-product2.15.0:
+
+```groovy
+gitVersion(prefix:'my-product') // -> 2.15.0
+gitVersion(prefix:'my-product', includePrefix:true) // -> my-product2.15.0
```

Comment:
For supporting both keeping the prefix and stripping it, I'm fine with just stripping it. Since you are grabbing the version for a certain product based on the prefix, returning just the version is acceptable to me.

--------------

What's the reason for so strongly trying to avoid the `v` prefix in git tags if it won't affect the published version? 

An example Lerna tag is:
@npmorg/product-api@X.Y.Z

I think it's possible to focus around a delimiter between the prefix and the version, so:
`.+[-\.@_]$`. This would allow the prefix to be whatever you want, but it avoids the v3.1.0 case. We may run into the occasional PR adding a delimiter to the regex.